src/globus_sdk/services/auth/client/confidential_client.py

```diff
@@ -447,6 +445,17 @@ def create_child_client(
                 "AuthClient.create_client requires either 'public_client' or "
                 "'client_type'."
             )
+        # terms_and_conditions and privacy_policy must both be set or unset
+        if bool(terms_and_conditions) ^ bool(privacy_policy):
+            raise exc.GlobusSDKUsageError(
+                "terms_and_conditions and privacy_policy must both be set or unset"
+            )
+        links: dict[str, str | MissingType] | MissingType = MISSING
+        if terms_and_conditions or privacy_policy:
```

Comment:
It's not incorrect to use an `or` here, but since we know from above that both or neither are set, it's sufficient to only check one.

I _definitely_ think it's more readable to show both variables on this line, rather than checking only one, but the `or` catches my eye because it will never evaluate if `terms_and_conditions` is set. Perhaps:

```suggestion
        if terms_and_conditions and privacy_policy:
```

Extra work, but it doesn't rely on readers understanding the XOR above to figure out a bare `if terms_and_conditions:`...